localstack/services/dynamodb/dynamodb_listener.py

```diff
@@ -769,6 +839,76 @@ def dynamodb_get_table_stream_specification(table_name):
         raise e
 
 
+def is_kinesis_stream_exists(stream_arn):
+    # connect to kinesis
+    kinesis = aws_stack.connect_to_service('kinesis')
+    # check if the stream exists in kinesis for the user
+    filtered = list(filter(lambda stream_name: stream_name == stream_arn.split('/', 1)[1],
+        kinesis.list_streams()['StreamNames']))
+    if filtered:
+        return True
+    return False
+
+
+def dynamodb_enable_kinesis_streaming_destination(data, table_def):
+    if (table_def.get('KinesisDataStreamDestinationStatus') in ['DISABLED', 'ENABLE_FAILED', None]):
+        table_def['KinesisDataStreamDestinations'] = table_def.get('KinesisDataStreamDestinations') or []
+        # remove the destination if already present
+        for dest in table_def['KinesisDataStreamDestinations']:
+            if dest['StreamArn'] == data['StreamArn']:
+                table_def['KinesisDataStreamDestinations'].remove(dest)
```

Comment:
I think this could result in a `list changed size during iteration` error at runtime. Better use list comprehension:
```
table_def['KinesisDataStreamDestinations'] = [t for t in table_def['KinesisDataStreamDestinations']
    if t['StreamArn'] != data['StreamArn']]
```